server/pulp/server/async/tasks.py

```diff
@@ -758,7 +758,11 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
             # celery will log the traceback
         if kwargs.get('scheduled_call_id') is not None:
             utils.increment_failure_count(kwargs['scheduled_call_id'])
-        if not self.request.called_directly:
+        try:
+            called_directly = self.request.called_directly
+        except Exception:
```

Comment:
can we catch a more specific exception?

Reply:
We could, but if we don't handle any exception from Celery by continuing to remove the locks Pulp will deadlock. I did rework this commit in two ways though:

1) to re-raise the Celery exception instead of using the `except Exception; pass` pattern
2) to also log the original exception since we're loosing that due to celery never logging it as we expect to.

Reply:
The changes make sense to me, thank you.